src/VisualStudio/Core/Def/StackTraceExplorer/StackTraceExplorerCommandHandler.cs

```diff
@@ -8,19 +8,70 @@
 using Microsoft.VisualStudio.LanguageServices.Setup;
 using Microsoft.VisualStudio.Shell;
 using Microsoft.VisualStudio.Shell.Interop;
-using Microsoft.VisualStudio.Text.Classification;
-using Microsoft.VisualStudio.Utilities;
 
 namespace Microsoft.VisualStudio.LanguageServices.StackTraceExplorer
 {
-    internal class StackTraceExplorerCommandHandler
+    internal class StackTraceExplorerCommandHandler : IVsBroadcastMessageEvents, IDisposable
     {
         private readonly RoslynPackage _package;
+        private readonly IThreadingContext _threadingContext;
         private static StackTraceExplorerCommandHandler? _instance;
+        private uint _vsShellBroadcastCookie;
 
         private StackTraceExplorerCommandHandler(RoslynPackage package)
         {
             _package = package;
+            _threadingContext = package.ComponentModel.GetService<IThreadingContext>();
+
+            var serviceProvider = (IServiceProvider)package;
+            var vsShell = serviceProvider.GetService(typeof(SVsShell)) as IVsShell;
+
+            if (vsShell is not null)
+            {
+                vsShell.AdviseBroadcastMessages(this, out _vsShellBroadcastCookie);
+            }
+        }
+
+        public int OnBroadcastMessage(uint msg, IntPtr wParam, IntPtr lParam)
+        {
+            if (msg != 0x001C) // WM_ACTIVATEAPP
+            {
+                return VSConstants.S_OK;
+            }
+
+            // wParam contains a value indicated if the window was activated.
+            // See https://docs.microsoft.com/en-us/windows/win32/winmsg/wm-activateapp 
+            // 1 = activate
+            // 0 = deactivated
+            if (wParam == IntPtr.Zero)
+            {
+                return VSConstants.S_OK;
+            }
+
+            var window = GetOrInitializeWindow();
+            var shouldActivate = _threadingContext.JoinableTaskFactory.Run(() => window.ShouldShowOnActivatedAsync(default));
```

Comment:
Does this need to block? Seems like this could operate on a background thread to do the analysis, and not block, and then just switch back to the UI thread to pop up the window later?